routers/api/v1/repo/pull_review.go

```diff
@@ -0,0 +1,520 @@
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "SubmitReview", err)
+		return
+	}
+
+	// convert response
+	apiReview, err := convert.ToPullReview(review, ctx.User)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "convertToPullReview", err)
+		return
+	}
+	ctx.JSON(http.StatusOK, apiReview)
+}
+
+// SubmitPullReview submit a pending review to an pull request
+func SubmitPullReview(ctx *context.APIContext, opts api.SubmitPullReviewOptions) {
+	// swagger:operation POST /repos/{owner}/{repo}/pulls/{index}/reviews/{id} repository repoSubmitPullReview
+	// ---
+	// summary: Submit a pending review to an pull request
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: index
+	//   in: path
+	//   description: index of the pull request
+	//   type: integer
+	//   format: int64
+	//   required: true
+	// - name: id
+	//   in: path
+	//   description: id of the review
+	//   type: integer
+	//   format: int64
+	//   required: true
+	// - name: body
+	//   in: body
+	//   required: true
+	//   schema:
+	//     "$ref": "#/definitions/SubmitPullReviewOptions"
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/PullReview"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+	//   "422":
+	//     "$ref": "#/responses/validationError"
+
+	review, pr, isWrong := prepareSingleReview(ctx)
+	if isWrong {
+		return
+	}
+
+	if review.Type != models.ReviewTypePending {
+		ctx.Error(http.StatusUnprocessableEntity, "", fmt.Errorf("only a pending review can be submitted"))
+		return
+	}
+
+	// determine review type
+	reviewType, isWrong := preparePullReviewType(ctx, pr, opts.Event, opts.Body)
+	if isWrong {
+		return
+	}
+
+	// if review stay pending return
+	if reviewType == models.ReviewTypePending {
+		ctx.Error(http.StatusUnprocessableEntity, "", fmt.Errorf("review stay pending"))
+		return
+	}
+
+	headCommitID, err := ctx.Repo.GitRepo.GetRefCommitID(pr.GetGitRefName())
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "GitRepo: GetRefCommitID", err)
+		return
+	}
+
+	// create review and associate all pending review comments
+	review, _, err = pull_service.SubmitReview(ctx.User, ctx.Repo.GitRepo, pr.Issue, reviewType, opts.Body, headCommitID)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "SubmitReview", err)
+		return
+	}
+
+	// convert response
+	apiReview, err := convert.ToPullReview(review, ctx.User)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "convertToPullReview", err)
+		return
+	}
+	ctx.JSON(http.StatusOK, apiReview)
+}
+
+func preparePullReviewType(ctx *context.APIContext, pr *models.PullRequest, event api.ReviewStateType, body string) (models.ReviewType, bool) {
```

Comment:
Can you add names to return variables?, it is hard to know what bool represents here.

Reply:
@lafriks made suggestion to remove those - I'll add a comment for documentation porpuse